.github/workflows/cypress-end-to-end-tests.yml

```diff
@@ -5,7 +5,56 @@ on:
     branches: [main]
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    # Set job outputs to values from filter step
+    outputs:
+      static_react_task: ${{ steps.filter.outputs.static_react_task }}
+      static_react_task_with_tips: ${{ steps.filter.outputs.static_react_task_with_tips }}
+      mnist: ${{ steps.filter.outputs.mnist }}
+      template: ${{ steps.filter.outputs.template }}
+      toxicity_detection: ${{ steps.filter.outputs.toxicity_detection }}
+      abstractions: ${{ steps.filter.outputs.abstractions }}
+      data_model: ${{ steps.filter.outputs.data_model }}
+      operations: ${{ steps.filter.outputs.operations }}
+      tools: ${{ steps.filter.outputs.tools }}
+      mephisto-task: ${{ steps.filter.outputs.mephisto-task }}
+      mephisto-worker-addons: ${{ steps.filter.outputs.mephisto-worker-addons }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - uses: dorny/paths-filter@v2
```

Comment:
This GitHub action is used to allow for path filtering on the job level. 

GitHub actions natively supports path filtering on the action level. This is too broad for this use-case as it would look something like examples/**.

This would mean that all the tests would run every time a file is changed in static_react_task.

This solution allows only for the static_react_task test to run when a file is changed in static_react_task